pkg/mock/fileUtils.go

```diff
@@ -256,6 +256,43 @@ func (f *FilesMock) FileRemove(path string) error {
 	return nil
 }
 
+// RemoveAll deletes the association of the given path with any content and records the removal of the path.
+// If the path has not been registered before, it returns an error.
+func (f *FilesMock) RemoveAll(path string) error {
+	if f.files == nil {
+		return fmt.Errorf("the file '%s' does not exist: %w", path, os.ErrNotExist)
+	}
+	absPath := f.toAbsPath(path)
+
+	dirExists, _ := f.DirExists(path)
+	if !dirExists {
+		return fmt.Errorf("the path '%s' does not exist: %w", path, os.ErrNotExist)
+	}
+
+	delete(f.files, absPath)
+	f.removedFiles = append(f.removedFiles, absPath)
+
+	// Make sure the parent directory still exists, if it only existed via this one entry
+	leaf := filepath.Base(absPath)
+	absPath = strings.TrimSuffix(absPath, f.Separator+leaf)
+	if absPath != f.Separator {
+		relPath := strings.TrimPrefix(absPath, f.Separator+f.CurrentDir+f.Separator)
+		dirExists, _ := f.DirExists(relPath)
+		if !dirExists {
+			f.AddDir(relPath)
+		}
+	}
+
+	return nil
```

Comment:
I think this will not suffice. Maybe the needs of the `FilesMock` have out-grown it's initial implementation, which is simply a map of directory and file paths. Imagine you do something like this:
```golang
f := FilesMock{}
f.AddFile("path/to/a", ...)
f.AddFile("path/to/b", ...)
f.AddFile("path/to/c", ...)
```

These are now three entries in the map `f.files`. There are no entries for the implicitly created directories, but this will return true:
```golang
assertTrue(t, f.DirExists("path"))
assertTrue(t, f.DirExists("path/to"))
```

When you do this...

```golang
f.RemoveAll("path/to")
```

...the expected result is that `f.files` now contains a single entry "path" which is of directory type. So the implementation of `RemoveAll` would have to find all entries that have the given parent path, remove them and then make sure a new entry is created as needed to keep the implicit directory, if necessary.